### DISCUSSION TITLE
$nextTick error when x-data is a child of x-if

### DISCUSSION THREAD

**Discussion by TreetopFlyer** [alpinejs/alpine#2662] [7. Bugs (archived)] [answered]
In the following example:
```
<script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>

<form x-data>
    <div x-data="$store.drilldown">
        <input x-model="primary" id="primary" type="checkbox"/>
        <label for="primary">Primary</label>
    </div>
    <hr>
    <template x-if="$store.drilldown.primary">
        <div x-data="$store.drilldown">
            <input x-model="secondary" id="secondary" type="checkbox"/>
            <label for="secondary">Secondary</label>
        </div>
    </template>
</form>

<script>
document.addEventListener( 'alpine:init', () => {
    Alpine.store( 'drilldown', { primary: false, secondary: false });
});
</script>
```
Repeatedly toggling "Primary" will cause multiple "Secondary" form controls to be created and throw `Uncaught TypeError: Cannot redefine property: $nextTick`

Either 1) removing the `x-data` directive inside the `<template>` and directly binding that input e.g. `<input x-model="$store.drilldown.secondary"...` or 2) using x-show instead of `<template x-if="...` fixes the problem.

**Comment by SimoTod** [comment, marked as answer]
Not sure it can be classified as a bug. It's an incorrect use of x-data + store. Each x-data should be an indipendent object $store should be used on a sub properties of x-data. It looks like it works because your store property is an object so initially alpine doesn't complain but with that code you are defining magic properties on the store variable itself so when alpine tries the second time it fails (x-show works because it never destroys the component but it's still wrong).

**Reply by TreetopFlyer** [reply]
OK thank you very much!
I misunderstood the role of x-data. I was ultimately looking for a way set local template "context" to some field in the store to sort of "DRY up" template markup e.g.
```
<!--  direct to store -->
<div x-bind:class="$store.form.name.errors.length ? 'errors' : ''">
    <input x-model="$store.form.name.input" id="form-name" type="text" />
    <label x-html="$store.form.name.label" for="form-name"></label>
    <template x-for="e in $store.form.name.errors"><p x-html="e"></p></template>
</div>

<!-- DRY "context" -->
<div x-data="$store.form.name" x-bind:class="errors.length ? 'errors' : ''">
    <input x-model="input" id="form-name" type="text" />
    <label x-html="label" for="form-name"></label>
    <template x-for="e in errors"><p x-html="e"></p></template>
</div>
```

putting store values into x-data *seemed* to work in certain cases but I will change my ways.

**Reply by SimoTod** [reply]
Yeah, basically the current tradeoff is that you can't have it as a full data context, you need, at least, something like `x-data="{foo: $store.foo}"`

I believe that the idea is that you bind all the properties you need explicitly.

**Reply by TreetopFlyer** [reply]
Oh that's a great solution thanks again. Why didn't I think of that...
